roseau/load_flow/models/loads/loads.py

```diff
@@ -16,562 +15,222 @@
-            },
-        }
-
-
-class DeltaImpedanceLoad(AbstractLoad):
-    """Constant delta impedance loads.
-
-    The equations are the same as in the constance admittance delta load implementation.
-    """
-
-    def __init__(self, id: Any, bus: Bus, z: Sequence[complex], **kwargs) -> None:
-        """ImpedanceLoad constructor.
+    def is_flexible(self) -> bool:
+        """Check if the load is flexible.
 
-        Args:
-            id:
-                The id of the load.
-
-            bus:
-                Bus to be attached to.
-
-            z:
-                List of impedance for each phase (Ohms).
-        """
-        if "n" in kwargs:
-            kwargs.pop("n")
-        super().__init__(id=id, n=3, bus=bus, **kwargs)
-        if len(z) != 3:
-            msg = f"Incorrect number of impedance: {len(z)} instead of 3"
-            logger.error(msg)
-            raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_Z_SIZE)
-
-        if isinstance(z, Quantity):
-            z = z.m_as("ohm")
-
-        for zi in z:
-            if np.isclose(zi, 0):
-                msg = f"An impedance for load {self.id!r} is null"
-                logger.error(msg)
-                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_Z_VALUE)
-
-        self.z = z
-
-    @ureg.wraps(None, (None, "ohm"), strict=False)
-    def update_impedance(self, z: Sequence[complex]) -> None:
-        """Change the admittances of the load
-
-        Args:
-            z:
-                The new impedance to set (Ohms).
+        Returns:
+            True if the load is flexible, False otherwise.
         """
-        if len(z) != 3:
-            msg = f"Incorrect number of impedance: {len(z)} instead of 3"
-            logger.error(msg)
-            raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_Z_SIZE)
-
-        for zi in z:
-            if np.isclose(zi, 0):
-                msg = f"An impedance for load {self.id!r} is null"
-                logger.error(msg)
-                raise RoseauLoadFlowException(msg=msg, code=RoseauLoadFlowExceptionCode.BAD_Z_VALUE)
-        self.z = z
+        return self.flexible_parameters is not None
 
-    #
-    # Json Mixin interface
-    #
-    def to_dict(self) -> dict[str, Any]:
-        load_type = "dz"
-        za, zb, zc = self.z
-        return {
-            "id": self.id,
-            "function": load_type,
-            "impedances": {
-                "za": [za.real, za.imag],
-                "zb": [zb.real, zb.imag],
-                "zc": [zc.real, zc.imag],
-            },
-        }
-
-
-class FlexibleLoad(AbstractLoad):
-    """A class to depict a flexible load i.e. a load with control."""
-
-    flexible_parameter_class: type[FlexibleParameter] = FlexibleParameter
+    def add_control(self, parameters: list[FlexibleParameter]) -> None:
+        """Make the load flexible load i.e. a load with control.
```

Comment:
Having a method here to add a control is complicated for the `engine` part, because the internal C++ element should be determined during the constructor, since the `PowerLoad` and the `FlexibleLoad` lead to different equations. I think a `parameters: Optional[list[FlexibleParameter]]` can be added to the constructor of `Load` instead of this method.